addon/mixins/in-viewport.js

```diff
@@ -64,164 +65,164 @@ export default Mixin.create({
   },
 
   _startListening() {
-    this._setInitialViewport(window);
-    this._addObserverIfNotSpying();
-    this._bindScrollDirectionListener(window, get(this, 'viewportScrollSensitivity'));
-
-    if (!get(this, 'viewportUseRAF')) {
-      get(this, 'viewportListeners').forEach((listener) => {
-        const { context, event } = listener;
-        this._bindListeners(context, event);
-      });
-    }
-  },
-
-  _addObserverIfNotSpying() {
-    if (!get(this, 'viewportSpy')) {
-      this.addObserver('viewportEntered', this, this._unbindIfEntered);
-    }
+    this._setInitialViewport();
+    // this._addObserverIfNotSpying();
+    // this._bindScrollDirectionListener(window, get(this, 'viewportScrollSensitivity'));
+
+    // if (!get(this, 'viewportUseRAF')) {
+    //   get(this, 'viewportListeners').forEach((listener) => {
+    //     const { context, event } = listener;
+    //     this._bindListeners(context, event);
+    //   });
+    // }
   },
 
-  _setViewportEntered(context = null) {
-    assert('You must pass a valid context to _setViewportEntered', context);
+  // _addObserverIfNotSpying() {
+  //   if (!get(this, 'viewportSpy')) {
+  //     this.addObserver('viewportEntered', this, this._unbindIfEntered);
+  //   }
+  // },
 
-    const element = get(this, 'element');
+  _setViewportEntered(/* context = null */) {
+    // assert('You must pass a valid context to _setViewportEntered', context);
 
-    if (!element) {
+    if (!this.element) {
       return;
     }
 
-    const $contextEl = $(context);
-    const boundingClientRect = element.getBoundingClientRect();
-
-    this._triggerDidAccessViewport(
-      isInViewport(
-        boundingClientRect,
-        $contextEl.innerHeight(),
-        $contextEl.innerWidth(),
-        get(this, 'viewportTolerance')
-      )
-    );
-
-    if (boundingClientRect && get(this, 'viewportUseRAF')) {
-      rAFIDS[get(this, 'elementId')] = window.requestAnimationFrame(
-        bind(this, this._setViewportEntered, context)
-      );
+    const { top, left, bottom, right } = this.viewportTolerance;
+    const options = {
+      root: null, 
+      rootMargin: `${top}px ${right}px ${bottom}px ${left}px`,
+      threshold: this.viewportScrollSensitivity
```

Comment:
With a current default of 1, this means the whole target has to be in the viewport before the callback is fired.  Also the range is from 0-1.0, so that would change for users.  Perhaps just make a whole new API it seems then??